plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerConnectorFacetImpl.java

```diff
@@ -0,0 +1,284 @@
+        );
+      }
+    }
+    if (config.httpsPort != null) {
+      try {
+        httpsConfig = new DockerConnectorConfiguration(getRepository().getName(), HttpScheme.HTTPS, config.httpsPort);
+        connectorRegistrar.addConnector(httpsConfig);
+      }
+      catch (Exception e) {
+        log.warn(
+            "Could not configure HTTPS connector on port {} for docker repository {}",
+            config.httpsPort, getRepository().getName(), e
+        );
+      }
+    }
+  }
+
+  @Override
+  protected void doStop() throws Exception {
+    if (httpConfig != null) {
+      connectorRegistrar.removeConnector(httpConfig);
+    }
+    if (httpsConfig != null) {
+      connectorRegistrar.removeConnector(httpsConfig);
+    }
+  }
+
+  @Override
+  public Integer getHttpPort() {
+    return config.httpPort;
+  }
+
+  @Override
+  public Integer getHttpsPort() {
+    return config.httpsPort;
+  }
+
+  /**
+   * Validate that port is not used as used as HTTP/HTTPS ports in another repository.
+   */
+  private ConstraintViolation<?> validatePortNotInUse(final String path, final Integer port) {
+    if (port != null) {
+      Iterable<Repository> repositories;
+      try {
+        repositories = repositoryManager.browse();
+      }
+      catch (IllegalStateException ignored) { //NOSONAR
+        // this will happen when facet is validated during startup of repository manager (load from storage)
+        // ignore and consider configurations loaded from storage as valid
+        repositories = Collections.emptyList();
+      }
+      for (Repository repository : repositories) {
+        if (DockerFormat.NAME.equals(repository.getFormat().getValue()) && !repository.equals(getRepository())) {
+          DockerConnectorFacet connectorFacet = repository.facet(DockerConnectorFacet.class);
+          if (port.equals(connectorFacet.getHttpPort()) || port.equals(connectorFacet.getHttpsPort())) {
+            return constraintViolationFactory.createViolation(
+                path, "Port must be unique (conflicts with repository'" + repository.getName() + "')"
+            );
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Validate that HTTP and HTTPS ports are not equal.
+   */
+  private ConstraintViolation<?>[] validateNotSamePort(final String httpPortPath,
+                                                       final Integer httpPort,
+                                                       final String httpsPortPath,
+                                                       final Integer httpsPort)
+  {
+    if (httpPort != null && Objects.equals(httpPort, httpsPort)) {
+      return new ConstraintViolation<?>[]{
+          constraintViolationFactory.createViolation(
+              httpPortPath, "HTTP and HTTPs ports must be different"
+          ),
+          constraintViolationFactory.createViolation(
+              httpsPortPath, "HTTP and HTTPs ports must be different"
+          )
+      };
+    }
+    return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty array instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)